src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -613,25 +613,6 @@ def build_binary(
         }
         api_calls = [*select_left.api_calls, *select_right.api_calls]
 
-        # This is a temporary workaround for query comparison. The query_id_place_holder
-        # field of Query be a random generated id if not provided, which could cause the
-        # comparison of two queries fail even if the sql and is_ddl_on_temp_object
-        # value is the same.
-        # TODO (SNOW-1570952): Find a uniform way for the query comparison
-        def _query_exists(current_query: Query, existing_queries: List[Query]) -> bool:
-            for existing_query in existing_queries:
-                if (
-                    (current_query.sql == existing_query.sql)
-                    and (
-                        current_query.is_ddl_on_temp_object
-                        == existing_query.is_ddl_on_temp_object
-                    )
-                    and (current_query.params == existing_query.params)
-                ):
```

Comment:
current `Query.__eq__` does not check for params equality. Is that still okay?

Reply:
i actually view this as a bug for query comparison, i just updated the query comparison to see if any test fails